### PR TITLE
Typescript loaders

### PR DESCRIPTION
**Pull request by Mad-Kat**
This PR adds the ability to import loaders as ESM & CJS.

As there is already a build step for all the other code, adding a build step for loaders doesn't add much overhead.

All built code will be generated in /dist/* to have a better distinction between built code and code one can manually change.

### CODE REVIEW

File: packages/next-yak/withYak/index.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ const addYak = (yakOptions: YakConfigOptions, nextConfig: NextConfig) => {
     });
     webpackConfig.module.rules.push({
       test: /\.yak\.module\.css$/,
-      loader: path.join(currentDir, "../loaders/cssloader.cjs"),
+      loader: path.join(currentDir, "../loaders/cssloader.js"),
```
Comment (Mad-Kat) [inline packages/next-yak/withYak/index.ts:33]: I don't know if we need to separate CJS & ESM loaders depending on the `withYak` type

File: packages/next-yak/tsup.config.ts [hunk 2]
```diff
@@ -89,8 +89,20 @@ export default defineConfig([
...
     target: "es2022",
-    outDir: "withYak",
+    outDir: "dist/withYak",
+  },
+  // loaders
+  {
+    entryPoints: ["loaders/cssloader.ts", "loaders/tsloader.ts"],
+    format: ["esm", "cjs"],
+    minify: false,
+    sourcemap: true,
```
[the remark this answers is not in the document]
Comment (jantimon) [reply packages/next-yak/tsup.config.ts:100]: evtl für stacktraces

File: packages/next-yak/withYak/index.ts (hunk 1 above)
Comment (jantimon) [reply packages/next-yak/withYak/index.ts:33]: no - webpack 5 is happy with cjs files

File: packages/next-yak/dist/withYak/index.cjs [hunk 3]
```diff
@@ -57,7 +57,7 @@ var addYak = (yakOptions, nextConfig) => {
     });
     webpackConfig.module.rules.push({
       test: /\.yak\.module\.css$/,
-      loader: import_node_path.default.join(currentDir, "../loaders/cssloader.cjs"),
+      loader: import_node_path.default.join(currentDir, "../loaders/cssloader.js"),
```
Comment (jantimon) [inline packages/next-yak/dist/withYak/index.cjs:60]: I am not sure if that breaks cjs support - did you try it in a cjs app?

**Review by jantimon — APPROVED** [review]

File: packages/next-yak/withYak/index.ts (hunk 1 above)
Comment (Mad-Kat) [reply packages/next-yak/withYak/index.ts:33]: [As it turns out ](https://stackoverflow.com/a/76556992) Webpack > 5.80.0 can use both CJS and ESM
I will change it to `.cjs` for maximal compatibility

**Review by jantimon — APPROVED** [review]

**Pull request merged**